app/components/blacklight/gallery/slideshow_preview_component.rb

```diff
@@ -7,7 +7,14 @@ class SlideshowPreviewComponent < Blacklight::DocumentComponent
 
       def initialize(document:, document_counter: nil, **args)
         super(document: document, document_counter: document_counter, **args)
-        @document_counter = document_counter || @counter
+        # ViewComponent 3, document_counter = 0
+        # ViewComponent 3, @counter = 1
+        # ViewComponent 3, COLLECTION_INDEX_OFFSET = 1
+
+        # ViewComponent 2, document_counter = 1
+        # ViewComponent 2, @counter = 1
+        # ViewComponent 2, COLLECTION_INDEX_OFFSET = 0
+        @document_counter = document_counter ? document_counter + (COLLECTION_INDEX_OFFSET - 1) : @counter - 1
```

Comment:
Is the problem here that the slide number is using `@document_counter`, but it should use `@counter` instead? Isn't this adjustment already done by  https://github.com/projectblacklight/blacklight/blob/d25f695fbfd377983afe15f90e011509e12e977f/app/components/blacklight/document_component.rb#L111 ?   Can we remove all of this code an rely solely on what Blacklight provides?

Reply:
I mean yeah that would work, we would have to do counter - 1 because we want to start at 0, not 1 which is what blacklight does. I figured there was probably a good reason to use document_counter first instead of @counter which is why I didn't change it.

Reply:
I think it's a bit dangerous to reset instance variables that are originally set by the parent class. I think it's much safer to create a new variable if you can't use `@counter`.

Reply:
Also I am pretty sure I am going to have to rewrite a bunch of tests to get this to work in vc2. It looks like vc_test_controller doesn't exist in vc2

Reply:
Figured it out.